src/roles/sdwan-portal-health/tasks/main.yml

```diff
@@ -0,0 +1,203 @@
+      delegate_to: localhost
+
+    - name: Write unexpected wsrep connected to report file
+      nuage_append: filename="{{ report_path }}" text="ERROR{{':'}} Expected wsrep_connected to be 'ON'. Got {{ wsrep_connected }}\n"
+      when: wsrep_connected is not match("ON")
+      delegate_to: localhost
+
+    - name: Set fact for wsrep local state
+      set_fact:
+        wsrep_local_state_comment: "{{ cluster_health_json['wsrep_local_state_comment'] }}"
+
+    - name: Write expected wsrep local state to report file
+      nuage_append: filename="{{ report_path }}" text="wsrep_local_state_comment{{':'}} {{ wsrep_local_state_comment }}\n"
+      when: wsrep_local_state_comment is match("Synced")
+      delegate_to: localhost
+
+    - name: Write unexpected wsrep local state to report file
+      nuage_append: filename="{{ report_path }}" text="ERROR{{':'}} Expected 'Synced' for wsrep_local_state_comment. Got {{ wsrep_local_state_comment }}\n"
+      when: wsrep_local_state_comment is not match("Synced")
+      delegate_to: localhost
+
+    - name: Set fact for wsrep ready
+      set_fact:
+        wsrep_ready: "{{ cluster_health_json['wsrep_ready'] }}"
+
+    - name: Write expected wsrep ready to report file
+      nuage_append: filename="{{ report_path }}" text="wsrep_ready{{':'}} {{ wsrep_ready }}\n"
+      when: wsrep_ready is match("ON")
+      delegate_to: localhost
+
+    - name: Write unexpected wsrep ready to report file
+      nuage_append: filename="{{ report_path }}" text="ERROR{{':'}} Expected wsrep_ready to be 'ON'. Got {{ wsrep_ready }}\n"
+      when: wsrep_ready is not match("ON")
+      delegate_to: localhost
+
+    - name: Set fact for wsrep cluster size
+      set_fact:
+        wsrep_cluster_size: "{{ cluster_health_json['wsrep_cluster_size'] }}"
+
+    - name: Write expected wsrep cluster size
+      nuage_append: filename="{{ report_path }}" text="wsrep_cluster_size{{':'}} {{ wsrep_cluster_size }}\n"
+      when: wsrep_cluster_size|int == 3
+      delegate_to: localhost
+
+    - name: Write out unexpected wsrep cluster size
+      nuage_append: filename="{{ report_path }}" text="Expected wsrep_cluster_size to be 3. Got {{ wsrep_cluster_size }}\n"
+      when: wsrep_cluster_size|int != 3
+      delegate_to: localhost
+
+    - name: Set fact for wsrep cluster status
+      set_fact:
+        wsrep_cluster_status: "{{ cluster_health_json['wsrep_cluster_status'] }}"
+
+    - name: Write expected wsrep cluster status to report file
+      nuage_append: filename="{{ report_path }}" text="wsrep_cluster_status{{':'}} {{ wsrep_cluster_status }}\n"
+      when: wsrep_cluster_status is match("Primary")
+      delegate_to: localhost
+
+    - name: Write unexpected wsrep cluster status to report file
+      nuage_append: filename="{{ report_path }}" text="ERROR{{':'}} Expected wsrep_cluster_status to be 'Primary'. Got {{ wsrep_cluster_status }}\n"
+      when: wsrep_cluster_status is not match("Primary")
+      delegate_to: localhost
```

Comment:
Looks like a lot of these tasks have delegate_to: localhost. Maybe they can be clubbed together in a block

Reply:
Done, thanks!